config/main.py

```diff
@@ -530,7 +530,99 @@ def is_ipaddress(val):
         return False
     return True
 
+# class for locking entire config process
+class ConfigDbLock():
+    def __init__(self):
+        self.lockName = "LOCK|configDbLock"
+        self.timeout = 10
+        self.pid = os.getpid()
+        self.client = None
+
+        self._acquireLock()
+        return
+
+    def _acquireLock(self):
+        try:
+            # connect to db
+            db_kwargs = dict()
+            configdb = ConfigDBConnector(**db_kwargs)
+            configdb.connect()
+
+            self.client = configdb.get_redis_client('CONFIG_DB')
+            # Set lock and expire time. Process may get killed b/w set lock and
+            # expire call.
+            if self.client.hsetnx(self.lockName, "PID", self.pid):
+                self.client.expire(self.lockName, self.timeout)
+            # if lock exists but expire timer not running, run expire time and
+            # abort.
+            elif not self.client.ttl(self.lockName):
+                click.echo(":::Unable to acquire lock. Resetting timer and aborting:::");
+                self.client.expire(self.lockName, self.timeout)
+                sys.exit(1)
+            else:
+                click.echo(":::Unable to acquire lock. Aborting:::");
+                sys.exit(1)
+        except Exception as e:
+            click.echo(":::Exception: {}:::".format(e))
+            sys.exit(1)
+        return
+
+    def reacquireLock(self):
+        try:
+            # Try to set lock first
+            if self.client.hsetnx(self.lockName, "PID", self.pid):
+                self.client.expire(self.lockName, self.timeout)
+            # if lock exists, check who owns it
+            else:
+                p = self.client.pipeline(True)
+                # watch, we do not want to work on modified lock
+                p.watch(self.lockName)
+                # if current process holding then extend the timer
+                if p.hget(self.lockName, "PID") == str(self.pid):
+                    self.client.expire(self.lockName, self.timeout)
+                    p.unwatch()
+                    return
+                else:
+                    # some other process is holding the lock.
+                    click.echo(":::Unable to reacquire lock (lock PID: {}, self.pid: {}):::".\
+                        format(p.hget(self.lockName, "PID"), self.pid))
+                    p.unwatch()
+                    sys.exit(1)
+
+        except Exception as e:
+            click.echo(":::Exception: {}:::".format(e))
+            sys.exit(1)
```

Comment:
This block isn't indented correctly. Can you please fix the indentation and remove the extra blank line above it?

Reply:
I can remove the extra line[591], but I think rest is indented correctly, i.e. 593, 594 are 4 space + from 592. 
Note line 595 is in function block.

Reply:
Thanks. I believe the extra blank line caused an optical illusion; it looked like the `except` statement was aligned with the function definition.